src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -708,7 +711,9 @@ public ConnectionDataFetcher(Class<? extends Connection> connection, DataFetcher
         @Override
         public Object get(DataFetchingEnvironment environment) {
```

Comment:
Please make the ConnectionDataFetcher generic (ConnectionDataFetcher<T> implements DataFetcher<T>)

Reply:
I'm not sure how to do that easily, as ConnectionDataFetcher is built from the connection class returned by the annotation, which is not generic. This would imply adding generics in annotation, DispatchingConnection, and maybe other places. In all cases changing ConnectionDataFetcher is not related to this ticket.